src/server/service/diagnostics.rs

```diff
@@ -0,0 +1,684 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use futures::Future;
+use grpcio::{RpcContext, UnarySink};
+use kvproto::diagnosticspb::{
+    Diagnostics, SearchLogRequest, SearchLogResponse, ServerInfoRequest, ServerInfoResponse,
+};
+
+use crate::server::Error;
+
+/// Service handles the RPC messages for the `Diagnostics` service.
+#[derive(Clone)]
+pub struct Service {
+    log_file: String,
+}
+
+impl Service {
+    pub fn new(log_file: String) -> Self {
+        Service { log_file }
+    }
+}
+
+impl Diagnostics for Service {
+    fn search_log(
+        &mut self,
+        ctx: RpcContext<'_>,
+        req: SearchLogRequest,
+        sink: UnarySink<SearchLogResponse>,
+    ) {
+        let future = log::search(&self.log_file, req)
+            .map_err(|err| Error::Other(box_err!("Search log error: {:?}", err)))
+            .and_then(|res| sink.success(res).map_err(Error::from))
+            .map_err(|e| debug!("Diagnostics rpc failed"; "err" => ?e));
+        ctx.spawn(future);
```

Comment:
The service is registered into same environment with `kv::Service`. So they share grpc threads. Is it better to move the logic into a thread pool? I think sharing the pool with `Debug` is ok.